pkg/core/resources/apis/mesh/meta_validator.go

```diff
@@ -3,33 +3,70 @@ package mesh
 import (
 	"regexp"
 
-	"github.com/kumahq/kuma/pkg/core/resources/model"
+	core_model "github.com/kumahq/kuma/pkg/core/resources/model"
 	"github.com/kumahq/kuma/pkg/core/validators"
 )
 
-var nameMeshRegexp = regexp.MustCompile("^[0-9a-z-_]*$")
+var (
+	backwardCompatRegexp = regexp.MustCompile(`^[0-9a-z-_.]*$`)
+	backwardCompatErrMsg = "invalid characters. Valid characters are numbers, lowercase latin letters and '-', '_', '.' symbols."
+)
+
+var (
+	identifierRegexp = regexp.MustCompile(`^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$`)
```

Comment:
Isn't there a max size too?

Reply:
If I understand https://github.com/kumahq/kuma/pull/6123/files correctly we rely on the store to validate the length

Reply:
Would it make sense to have api level validation? Or just a recipe for disaster?

Reply:
We don't have issues with validating name length at this moment (it's already compatible with Kubernetes), so I don't think it's necessary to include in this PR. But I can add it, should be a small change

Reply:
+1 to add it

Reply:
Added. We actually handled this pretty badly on the client side:
```
$ kumactl apply -f resource_with_long_name.yaml
Error: Could not create a resource (Internal Server Error)
```